### PR TITLE
Support dump pass name and disable specified passes

### PR DESCRIPTION
**Pull request by linqun**

### CODE REVIEW

File: util/llpcPassManager.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "llvm/Support/CommandLine.h"
 
 #include "llpcPassManager.h"
+#include "llpcDebug.h"
```
Comment (amdrexu) [inline util/llpcPassManager.cpp]: Alphabet order. Move it up.

File: util/llpcPassManager.cpp [hunk 2]
```diff
@@ -49,6 +50,12 @@ static cl::opt<std::string> DumpCfgAfter("dump-cfg-after",
                                          cl::desc("Dump CFG as .dot files after specified pass"),
                                          cl::init(""));
 
+// -dump-pass-name : dump executed pass name
+static cl::opt<bool> DumpPassName("dump-pass-name", cl::desc("Dump executed pass name"), cl::init(false));
+
+// -disable-pass-index: indices of passes to be disabled
```
Comment (amdrexu) [inline util/llpcPassManager.cpp]: -index -> -indices

File: util/llpcPassManager.cpp [hunk 3]
```diff
@@ -49,6 +50,12 @@ static cl::opt<std::string> DumpCfgAfter("dump-cfg-after",
...
+static cl::list<uint32_t> DisablePassIndices("disable-pass-index", cl::ZeroOrMore, cl::desc("Indices of passes to be disabled"));
```
Comment (amdrexu) [inline util/llpcPassManager.cpp]: -index -> -indices

File: util/llpcPassManager.cpp [hunk 4]
```diff
@@ -107,12 +120,33 @@ void Llpc::PassManager::add(
...
     // Skip the jump threading pass as it interacts really badly with the structurizer.
-    if (pPass->getPassName().equals("Jump Threading"))
+    if (passId == m_jumpThreading)
     {
         return;
     }
 
+    if (passId != m_printModule)
+    {
+        uint32_t passIdx = (*m_pPassIndex)++;
```
Comment (amdrexu) [inline util/llpcPassManager.cpp]: passIdx -> passIndex

Try to keep the naming similar

**Comment by trenouf** [comment]
I can see why you want this for experimenting with which passes we use, but should we really have it checked in?

**Comment by amdvlk-admin** [comment]
### Test summary for commit 687e36a3d2c141daf8d4b735eac90db795e101f8
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 99901922d67e502ff4233a8bfcbfa0347c0ffa1b</li>
<li>SPVGEN: dev 69311f7c1e220d190909298b9c7bd702ecc237ff</li>
<li>XGL: dev 9795e006e9adb7b49b8abd32cd47c49b290fa3fc</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 901/901 (100.0%)</li>
<li>Failed: 0/901 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by linqun** [comment]
> I can see why you want this for experimenting with which passes we use, but should we really have it checked in?

as you know, the major purpose of this change is to select optimization pass. but I think it is also useful when we debug some quality issues. we can disable some suspicious pass without modify any code by this option. and It is a minor change, I don't think it has some side effect. please let me know if you have any concerns about it.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 12b8b7256b844aa83dc752866042f027b49cedd9
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 99901922d67e502ff4233a8bfcbfa0347c0ffa1b</li>
<li>SPVGEN: dev 69311f7c1e220d190909298b9c7bd702ecc237ff</li>
<li>XGL: dev 9795e006e9adb7b49b8abd32cd47c49b290fa3fc</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 901/901 (100.0%)</li>
<li>Failed: 0/901 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdrexu** [comment]
Please leave this commit unmerged until Tim finishes discussing it with Qun.

**Comment by trenouf** [comment]
OK, please go ahead.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9b285995997afd852e75948e915fa9b6123eaf5a
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 99901922d67e502ff4233a8bfcbfa0347c0ffa1b</li>
<li>SPVGEN: dev 5378c7618a2a3bf60ba01a86db980930e8581146</li>
<li>XGL: dev 9795e006e9adb7b49b8abd32cd47c49b290fa3fc</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 901/901 (100.0%)</li>
<li>Failed: 0/901 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**